drivers/sensor/lsm6dsr/Kconfig

```diff
@@ -0,0 +1,125 @@
+# LSM6DSR accelerometer and gyroscope sensor configuration
+# options
+
+# Copyright (c) 2023, Andrei Hutanu
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig LSM6DSR
+	bool "LSM6DSR I2C accelerometer and gyroscope Chip"
+	default y
+	depends on DT_HAS_ST_LSM6DSR_ENABLED
+	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_ST_LSM6DSR),i2c)
+	help
+	  Enable driver for LSM6DSR accelerometer and gyroscope
+	  sensor.
+
+if LSM6DSR
+
+choice LSM6DSR_TRIGGER_MODE
+	prompt "Trigger mode"
+	help
+	  Specify the type of triggering to be used by the driver.
+
+config LSM6DSR_TRIGGER_NONE
+	bool "No trigger"
+
+config LSM6DSR_TRIGGER_GLOBAL_THREAD
+	bool "Use global thread"
+	depends on GPIO
+	select LSM6DSR_TRIGGER
+
+config LSM6DSR_TRIGGER_OWN_THREAD
+	bool "Use own thread"
+	depends on GPIO
+	select LSM6DSR_TRIGGER
+
+endchoice
+
+config LSM6DSR_TRIGGER
+	bool
+
+config LSM6DSR_THREAD_PRIORITY
+	int "Thread priority"
+	depends on LSM6DSR_TRIGGER_OWN_THREAD
+	default 10
+	help
+	  Priority of thread used by the driver to handle interrupts.
+
+config LSM6DSR_THREAD_STACK_SIZE
+	int "Thread stack size"
+	depends on LSM6DSR_TRIGGER_OWN_THREAD
+	default 1024
+	help
+	  Stack size of thread used by the driver to handle interrupts.
+
+menu "Attributes"
+
+config LSM6DSR_GYRO_FS
+	int "Gyroscope full-scale range"
+	default 0
+	help
+	  Specify the default gyroscope full-scale range.
+	  An X value for the config represents a range of +/- X degree per
+	  second. Valid values are:
+	  0:  Full Scale selected at runtime
+	  125:   +/- 125dps
+	  250:   +/- 250dps
+	  500:   +/- 500dps
+	  1000:  +/- 1000dps
+	  2000:  +/- 2000dps
+	  4000:  +/- 4000dps
```

Comment:
Please move xl/gyro fs/odr setting into DT. See here: https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/sensor/st%2Clsm6dso-common.yaml (included in the same dir by i2c and spi yaml files)